Projects/PixelGameEngine/PixelTutorials/Pathfinding/main.cpp

```diff
@@ -0,0 +1,356 @@
+	{
+		for (int x = 0; x < MapWidth; x++)
+		{
+			for (int y = 0; y < MapHeight; y++)
+			{
+				//Either of these can be commented out. Commenting out both technically also works.
+
+				ConnectOrthogonally(x, y);
+
+				ConnectDiagonally(x, y);
+				
+			}
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void ConnectOrthogonally(const int x, const int y)
+	{
+		if (y > 0)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y - 1) * MapWidth + (x + 0)]);
+		}
+		if (y < MapHeight - 1)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y + 1) * MapWidth + (x + 0)]);
+		}
+		if (x > 0)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y + 0) * MapWidth + (x - 1)]);
+		}
+		if (x < MapWidth - 1)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y + 0) * MapWidth + (x + 1)]);
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void ConnectDiagonally(const int x, const int y)
+	{
+		if (y > 0 && x > 0)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y - 1) * MapWidth + (x - 1)]);
+		}
+		if (y < MapHeight - 1 && x > 0)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y + 1) * MapWidth + (x - 1)]);
+		}
+		if (y > 0 && x < MapWidth - 1)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y - 1) * MapWidth + (x + 1)]);
+		}
+		if (y < MapHeight - 1 && x < MapWidth - 1)
+		{
+			Nodes[y * MapWidth + x].Neighbors.push_back(&Nodes[(y + 1) * MapWidth + (x + 1)]);
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void UpdateGridStates()
+	{
+		//Use integer division to nicely get cursor position in node space
+		int SelectedNodeX = GetMouseX() / NodeSize;
+		int SelectedNodeY = GetMouseY() / NodeSize;
+
+		if (GetMouse(0).bReleased) //Use mouse to draw maze, shift and ctrl to place start and end
+		{
+			if (SelectedNodeX >= 0 && SelectedNodeX < MapWidth)
+			{
+				if (SelectedNodeY >= 0 && SelectedNodeY < MapHeight)
+				{
+					if (GetKey(olc::Key::SHIFT).bHeld)
+					{
+						StartNode = &Nodes[SelectedNodeY * MapWidth + SelectedNodeX];
+					}
+					else if (GetKey(olc::Key::CTRL).bHeld)
+					{
+						EndNode = &Nodes[SelectedNodeY * MapWidth + SelectedNodeX];
+					}
+					else
+					{
+						Nodes[SelectedNodeY * MapWidth + SelectedNodeX].bObstacle = !Nodes[SelectedNodeY * MapWidth + SelectedNodeX].bObstacle;
+					}
+
+					SolveAStar();
+
+				}
+			}
+		}
+	}
+
+	//-------------------------------------------------------------------------------------------
+	void SolveAStar()
+	{
+		//Reset Navigation Graph - default all node states
+		ResetNavGraph();
+
+		//Lambda Functions, for convenience
+		auto Distance = [](FNode* a, FNode* b) 
+		{
+			return sqrtf((a->x - b->x) * (a->x - b->x) + (a->y - b->y) * (a->y - b->y));
+		};
+
+		auto Heuristic = [Distance](FNode* a, FNode* b)
```

Comment:
I don't know about these parameter names, should it be Uppercase? The same for the cases where you use x and y?
What does @JD say about this?